sdv/constraints/tabular.py

```diff
@@ -293,31 +327,31 @@ def _get_is_datetime(self, table_data):
 
         return is_datetime
 
+    def _check_columns_exist(self, table_data, field):
+        values = getattr(self, f'_{field}')
+        exists = [column not in table_data.columns for column in values]
+        if any(exists):
+            raise KeyError(f'The {field} columns {list(compress(values, exists))} '
+                           f'were not found in table_data. If {field} is a scalar, '
+                           f'set `scalar="{field}"`.')
+
     def _fit(self, table_data):
         """Learn the dtype of the high column.
 
         Args:
             table_data (pandas.DataFrame):
                 The Table data.
         """
-        if self._high_is_scalar is None:
-            self._high_is_scalar = self._high not in table_data.columns
-        if self._low_is_scalar is None:
-            self._low_is_scalar = self._low not in table_data.columns
-
-        if self._high_is_scalar and self._low_is_scalar:
-            raise TypeError('`low` and `high` cannot be both scalars at the same time')
-        elif self._low_is_scalar:
-            self.constraint_columns = (self._high,)
-            self._dtype = table_data[self._high].dtype
-        elif self._high_is_scalar:
-            self.constraint_columns = (self._low,)
-            self._dtype = table_data[self._low].dtype
-        else:
-            self._dtype = table_data[self._high].dtype
-
-        self._column_to_reconstruct = self._get_column_to_reconstruct()
-        self._diff_column = self._get_diff_column_name(table_data)
+        if self._scalar == 'high':
```

Comment:
this can be made simpler by negating:

```python3
if self._scalar != 'high':    # <- So high must be columns...
    self._check_columns_exists(table_data, 'high')
if self._scalar != 'low':
    self._check_columns_exists(table_data, 'low')
```